miner/miner.go

```diff
@@ -119,6 +121,42 @@ func New(eth Backend, config Config, engine consensus.Engine) *Miner {
 		lifeCtxCancel: cancel,
 		lifeCtx:       ctx,
 	}
+
+	// OP-Stack: Start background RPC polling
+	miner.startBackgroundInteropFailsafeDetection()
+
+	return miner
+}
+
+// OP-Stack: startBackgroundInteropFailsafeDetection starts a background goroutine that periodically
+// calls the supervisor over RPC to check if the failsafe is enabled
+func (miner *Miner) startBackgroundInteropFailsafeDetection() {
+	backend, ok := miner.backend.(BackendWithInterop)
+	if !ok {
+		log.Warn("Miner backend does not implement BackendWithInterop, skipping interop failsafe detection")
+		return
+	}
+
+	go func() {
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
+
+		log.Info("Starting background interop failsafe detection", "interval", "1s")
+
+		for {
+			select {
+			case <-ticker.C:
+				ctx, cancel := context.WithTimeout(miner.lifeCtx, 1*time.Second)
+				defer cancel()
+				// Simply calling CheckAccessList will update the failsafe status in the backend
+				// We do not need to do anything with the result
+				_ = backend.CheckAccessList(ctx, []common.Hash{}, interoptypes.CrossUnsafe, interoptypes.ExecutingDescriptor{})
```

Comment:
A little odd to use stubbed values here. There is an Admin API `admin_getFailsafeEnabled` that would return the value without having to provide stubbed values.

Reply:
Agreed. Made that change here [68bb732](https://github.com/ethereum-optimism/op-geth/pull/636/commits/68bb73204d0666c22b67408193a292c0c1b436b0). 

The only thing giving me pause is that we now need to use the _admin_ API which seems to suggest elevated privileges. But the getter itself ought not to need elevated privileges.

Reply:
I mean, the implementation at the moment is such that it _is_ in the admin namespace. And it might be a bit ugly to move the getter out of that namespace. But logically, it doesn't seem to me like it should require privileges to call this getter. It is already a part of the public API because we return a specific error from non-admin API calls which indicates whether failsafe is enabled. 

I think we have three ways forward:
1. infer failsafe indirectly through `CheckAccessList`, stubbing the parameters out
2. infer through the dedicated getter thru the admin API
3. move or replicate the getter in the non-admin API namespace of the supervisor, and query that from op-geth